include/shark/Algorithms/Trainers/CSvmTrainer.h

```diff
@@ -251,6 +266,9 @@ class CSvmTrainer : public AbstractSvmTrainer<
 			typedef SvmShrinkingProblem<SVMProblemType> ProblemType;
 			ProblemType problem(svmProblem,base_type::m_shrinking);
 			QpSolver< ProblemType > solver(problem);
+			if (svm.alpha().size1() > 0 && svm.alpha().size2() == 1) {
+				problem.setInitialSolution(blas::column(svm.alpha(), 0));
```

Comment:
What you hint at is a serious problem with the current interface. Changing C is relatively easy to deal with. We have to loop over the vector anyway, so checking the constraints is nearly for free. But what if the data set changes, i.e., the basis of the kernel expansion differs from the input portion of the LabeledData object passed to the train function? I could check for equality of the shared pointers inside the data containers - do we have any interface to check whether two data objects share data? We have data.makeIndependent(), but what I need here is data.isSharedWith(otherData). Any objections to adding such a function?